rust-toolchain.toml

```diff
@@ -1,2 +1,2 @@
 [toolchain]
-channel = "1.72.1"
+channel = "1.76.0"
```

Comment:
I guess this was changed because of:

```
error: package `clap_lex v0.7.0` cannot be built because it requires rustc 1.74 or newer, while the currently active rustc version is 1.72.1
```

```
clap_lex v0.7.0
└── clap_builder v4.5.2
    └── clap v4.5.3
        └── criterion v0.5.1
            └── aws-smithy-cbor v0.0.0-smithy-rs-head (/local/home/davidpz/workplace/smithy-ws/src/SmithyRsSource/rust-runtime/aws-smithy-cbor)
                ├── aws-smithy-http-server v0.0.0-smithy-rs-head (/local/home/davidpz/workplace/smithy-ws/src/SmithyRsSource/rust-runtime/aws-smithy-http-server)
                │   └── rpcv2 v0.0.1 (/local/home/davidpz/workplace/smithy-ws/src/SmithyRsSource/codegen-server-test/build/smithyprojections/codegen-server-test/rpcV2/rust-server-codegen)
                └── rpcv2 v0.0.1 (/local/home/davidpz/workplace/smithy-ws/src/SmithyRsSource/codegen-server-test/build/smithyprojections/codegen-server-test/rpcV2/rust-server-codegen)
```

Let's change it to 1.74.1, [which is what we currently have in `main`](https://github.com/smithy-lang/smithy-rs/blob/2d931fa8dc0da530fa0e17ec99d293895583bf0b/rust-toolchain.toml#L2), which should suffice. Rustc upgrades should go in a separate PR and usually require further changes.